_posts/2024-04-15-video-segmentation.md

```diff
@@ -0,0 +1,176 @@
+
+My main intention was to focus solely on the specific topic I was interested in. This is Video instance Segmentation (VIS) which extends the image instance segmentation task from the image domain to the video domain. The goal is to segment object instances from a predefined category set in videos and associate the instance identities across frames. It can also be seen as a combination of instance segmentation and object tracking in videos.
+
+As usually I will be trying to update this post with the most recent papers and code implementations I find interesting.
+
+## Introduction to Video Segmentation
+
+Video segmentation condenses a lot of different tasks which can have multiple names. I personaly like the taxonomy used in the [Youtube-VOS Dataset](https://youtube-vos.org/dataset/) which is one of the main benchmarks in this field so I will stick with it through this post. The different tasks are:
+
+- **Video Object Segmentation (VOS)**: targets at segmenting a particular object instance throughout the entire video sequence given only the object mask of the first frame.
+- **Video Instance Segmentation (VIS)**: extends image instance segmentation to videos, aiming to segment and track object instances across frames.
+- **Referring Video Object Segmentation (RVOS)**: is a task that requires to segment a particular object instance in a video given a natural language expression that refers to the object instance.
+
+<div class="post-center-image">
+<a href="/assets/svg/posts/2024-04-15-video-segmentation/video-segmentation-types-diagram.svg" class="lightbox-image" title="Diagram with the different video segmentation methods">
+ <img src="/assets/svg/posts/2024-04-15-video-segmentation/video-segmentation-types-diagram.svg" loading="lazy" alt="Diagram with the different video segmentation methods">
+</a>
+</div>
+
+{:refdef: class="image-caption"}
+*Diagram with the different video segmentation methods*
+{: refdef}
+
+
+## Key concepts
+
+### Input sequence length
+
+The input sequence length is crucial in video segmentation. Longer sequences provide more context for accurately segmenting objects, even during occlusion and appearance changes. However, they require more computational power and increase training and inference times. Think that transformers have a quadratic complexity with respect to the sequence length so most of the models we are going to discuss are trained with very short clips (mostly between 2 and 8 frames).
+
+### Stride
+
+The input sequence length defines the number of frames processed in parallel. The stride is the one that determines the temporal distance between frames in the input sequence. A stride of 1 means that the input sequence is a continuous sequence of frames, while a stride of 2 means that every other frame is skipped. By increasing the stride the system can work faster because it's looking at fewer frames.
```

Comment:
I would rephrase for clarity

```suggestion
The input sequence length defines the number of frames processed in parallel. The stride is the one that determines the temporal distance between adjacent frames in the input sequence. A stride of 1 means that the input sequence is a continuous sequence of frames, while a stride of 2 means that every other frame is skipped. By increasing the stride, the system can work faster because even though it will be looking at the same amount of frames at once, it will need to process fewer frames in total.
```